app/src/main/java/nodomain/freeyourgadget/gadgetbridge/database/ActivityDatabaseHandler.java

```diff
@@ -31,22 +30,35 @@ public ActivityDatabaseHandler(Context context) {
 
     @Override
     public void onCreate(SQLiteDatabase db) {
-        String CREATE_GBACTIVITYSAMPLES_TABLE = "CREATE TABLE " + TABLE_GBACTIVITYSAMPLES + "("
-                + KEY_ID + " INTEGER PRIMARY KEY,"
+        String CREATE_GBACTIVITYSAMPLES_TABLE = "CREATE TABLE " + TABLE_GBACTIVITYSAMPLES + " ("
                 + KEY_TIMESTAMP + " INT,"
                 + KEY_PROVIDER + " TINYINT,"
                 + KEY_INTENSITY + " SMALLINT,"
                 + KEY_STEPS + " TINYINT,"
                 + KEY_TYPE + " TINYINT,"
-                + " UNIQUE (" + KEY_TIMESTAMP + "," + KEY_PROVIDER + ") ON CONFLICT REPLACE)";
+                + " PRIMARY_KEY (" + KEY_TIMESTAMP + "," + KEY_PROVIDER + ") ON CONFLICT REPLACE)";
```

Comment:
A further optimization would be to add WITHOUT_ROWID to the CREATE TABLE statement. That would only work with Lollipop and upwards.

https://sqlite.org/withoutrowid.html
http://stackoverflow.com/questions/2421189/version-of-sqlite-used-in-android